test/OpenTelemetry.Tests/Shared/SkipOnWindowsRunnerTheoryAttribute.cs

```diff
@@ -0,0 +1,36 @@
+// <copyright file="SkipOnWindowsRunnerTheoryAttribute.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using Xunit;
+
+namespace OpenTelemetry.Tests;
+
+/// <summary>
+/// The GitHub-hosted Windows runners do not support Linux containers. This <see cref="TheoryAttribute" /> skips tests running on Windows runners.
+/// </summary>
+internal class SkipOnWindowsRunnerTheoryAttribute : TheoryAttribute
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SkipOnWindowsRunnerTheoryAttribute" /> class.
+    /// </summary>
+    public SkipOnWindowsRunnerTheoryAttribute()
+    {
+        if ("Windows".Equals(Environment.GetEnvironmentVariable("RUNNER_OS"), StringComparison.OrdinalIgnoreCase))
```

Comment:
For tests with this attribute, what will happen when I run them locally?

Reply:
Hm, it looks like `RUNNER_OS` does not return what the [docs](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) says, I will check.

Reply:
Ideally, the attribute causes the test to be skipped in all scenarios where either the environment is not compatible (windows, mac) or docker is not running.

Reply:
The only two cases are Docker is not running or Windows engine is enabled (very rare, macOS is supported too). If you like I can add something like we are doing in Testcontainers. We check if the Docker CLI response with the correct [platform](https://github.com/testcontainers/testcontainers-dotnet/blob/710987a2562f7bb5a58d64bb74cba17cc8681db4/tests/Testcontainers.Commons/DockerCli.cs#L44-L48). This will only enable the tests for environments that are capable of. WDYT?

Reply:
Lets see if the recent changes work better. BTW, I was trying a few times to test the changes in advance (on CI too), but I have serious issues building the forked version. I run into [this](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3412) issue all the time.

Reply:
You'll probably need fetch tags to your fork. This should allow minver to properly version assemblies.

Reply:
Ok, I think we need to move the tests to another class, because the remaining tests trigger xUnit.net's `IAsyncLifetime` again. That's why we see the issues on Windows.

Reply:
Gotcha, yea moving the this one test to something like `SqlClientIntegrationTests.cs` would be fine.

Alternatively, instead of using IAsyncLifetime, could you just do the following in this one test?

```csharp
public async Task SuccessfulCommandTest(...)
{
    await this.databaseContainer.StartAsync();
    ...
    await this.databaseContainer.DisposeAsync().AsTask();
}
```